cherrybomb-engine/src/lib.rs

```diff
@@ -23,44 +23,51 @@ fn verbose_print(config: &Config, required: Option<Verbosity>, message: &str) {
 
 pub async fn run(config: &Config) -> anyhow::Result<Value> {
     verbose_print(config, None, "Starting Cherrybomb...");
-
-    // Reading OAS file to string
     verbose_print(config, None, "Opening OAS file...");
-    let oas_file = match std::fs::read_to_string(&config.file) {
-        Ok(file) => file,
-        Err(e) => {
-            return Err(anyhow::anyhow!("Error reading OAS file: {}", e));
-        }
-    };
 
-    // Parsing OAS file to JSON
-    verbose_print(config, None, "Parsing OAS file...");
-    let oas_json: Value = match serde_json::from_str(&oas_file) {
-        Ok(json) => json,
-        Err(e) => {
-            return Err(anyhow::anyhow!("Error parsing OAS file: {}", e));
-        }
-    };
-
-    // Parsing JSON to OAS struct
-    verbose_print(config, Some(Verbosity::Debug), "Creating OAS struct...");
-    let oas: OAS3_1 = match serde_json::from_value(oas_json.clone()) {
-        Ok(oas) => oas,
-        Err(e) => {
-            return Err(anyhow::anyhow!("Error creating OAS struct: {}", e));
+    match config.file.extension() {
+        Some(ext) => {
+            verbose_print(config, None, "Reading OAS file...");
+            let oas_file = match std::fs::read_to_string(&config.file) {
+                Ok(file) => file,
+                Err(e) => return Err(anyhow::Error::msg(format!("Error opening OAS file: {}", e))),
+            };
+            let oas_json: Value = match ext.to_str() {
+                Some("json") => {
+                    verbose_print(config, None, "Parsing OAS file...");
+                    match serde_json::from_str(&oas_file) {
+                        Ok(json) => json,
+                        Err(e) => return Err(anyhow::Error::msg(format!("Error parsing OAS file: {}", e))),
+                    }
+                }
+                Some("yaml") => {
```

Comment:
this should also support `yml`.
However extension isnt really a good way to do this.
trying json, and falling back to yaml is another approach. c.f. https://github.com/oxidecomputer/progenitor/pull/227
my preferred approach is to look at the first non-whitespace char to see if it is json, e.g. https://github.com/paperclip-rs/paperclip/blob/43cfea1/src/v2/mod.rs#L110

Reply:
@GuyL99 Yes serde_yaml::Value and serde_json::Value can be used here without any trouble . 
But which approach do you prefer by read the file itself or only read the extension ?
And then please approve my PR :)